eventlet/green/ssl.py

```diff
@@ -40,59 +53,56 @@ class GreenSSLSocket(_original_sslsocket):
     settimeout(), and to close/reopen the connection when a timeout
     occurs at an unexpected juncture in the code.
     """
+    def __new__(cls, sock=None, keyfile=None, certfile=None,
+                server_side=False, cert_reqs=CERT_NONE,
+                ssl_version=PROTOCOL_SSLv23, ca_certs=None,
+                do_handshake_on_connect=True, *args, **kw):
+        if sys.version_info < (3, 7):
+            return super(GreenSSLSocket, cls).__new__(cls)
+        else:
```

Comment:
I'm a little uneasy about the fact that the default value for `sock` is `None`. I realize you copied that argument list from `ssl.SSLSocket.__init__()`, but I don't know under what circumstances we could actually arrive at this point with `sock is None`. Is that case supported by `ssl.SSLSocket`? Passing `None` to `GreenSocket` doesn't seem likely to be rewarding -- or trying to reference `None.fd`, etc.
I think what I'm asking is that you either defend against that case, or explain in comments why that won't happen.

Reply:
This is the orginal SSLSocket.__init__ signature. 

```
In [157]: ssl.SSLSocket.__init__
Out[157]: <function ssl.SSLSocket.__init__(self, sock=None, keyfile=None, certfile=None, server_side=False, cert_reqs=<VerifyMode.CERT_NONE: 0>, ssl_version=<_SSLMethod.PROTOCOL_TLS: 2>, ca_certs=None, do_handshake_on_connect=True, family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=0, fileno=None, suppress_ragged_eofs=True, npn_protocols=None, ciphers=None, server_hostname=None, _context=None, _session=None)>
```